text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,151 @@
+
+### Specification requirements
+
+The specification should declare all requirements for a specified component to be declared stable. When implementations are still missing features required by the specification, they can be at most Alpha status.
+
+### Tier-1 SIGs
+
+This OTEP recommends the following SIGs to be included as part of the Tier-1 SIGs for OpenTelemetry:
+
+* Spec General
+* Semantic Conventions (WG)
+* Java
+* Java auto-instrumentation
+* Go
+* .Net
+* JavaScript
+
+Once this OTEP is approved, the TC and GC members will vote on the individual SIGs from the list above and are free to make further suggestions. A final list of Tier-1 SIGs shall be in the "community" repository.
+
+### Tier-1 Components
+
+When discussing the components' maturity, we often debate what should be included in the evaluation criteria. To help clarify that, we introduce the concept of "Tier-1 SIGs" and "Tier-1 Components". Tier-1 Components are critical for a specific SIG: only when all Tier-1 Components are stable can a project be evaluated as stable. For instance, the "OTLP receiver" for the OpenTelemetry Collector could potentially be a "Tier-1 Component". Only after this component is declared stable may the Collector be declared stable, provided all other Tier-1 Components are also stable.
+
+Projects MUST list their Tier-1 Components in the main readme file. A Tier-1 Component MAY be hosted under the project's "contrib" repository. Not all components under the main repository (sometimes called the "core" repository) are automatically Tier-1 Components.
+
+A component can be declared "Tier-1" only if they have the following characteristics:
+
+* The component has at least the same maturity level as the SIG
+* The component has a clear code owner
+* The code owner is active within the project
+* The maintainers of the project are comfortable supporting the component should the code owner decide to step down or leave the project
+
+### Tier-1 Signals
+
+Similar to a Tier-1 Component, a Tier-1 Signal must have stable support in order to declare OpenTelemetry as stable. Declaring a signal as Tier-1 means that all projects must also see that signal as required for declaring the project stable.
+
+Not all signals will have the same requirements. For instance, Logs might not provide an instrumentation API specification.
+
+List of Tier-1 signals and their capabilities:
+
+* Logs
+  * Data model
+  * SDK specification
+  * Instrumentation SDKs
+  * Bridge API specification
+* Metrics
+  * Data model
+  * API specification
+  * Instrumentation APIs
+  * SDK specification
+  * Instrumentation SDKs
+* Traces
+  * Data model
```

Comment:
If only we had a trace data mode 😢

Reply:
Your comment intrigued me. Indeed, we don't seem to have a "data model" page under our Tracing spec, but what I would expect to see there is available as part of the API, like this: https://opentelemetry.io/docs/specs/otel/trace/api/#span

What I had in mind though was this: https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto

In any case, do you believe there's something we should address in this PR here?

Reply:
No, probably nothing to address in this PR.